### ISSUE TITLE
[avx512] General AVX-512 tracking issue

### ISSUE THREAD

**Issue by Lokathor** [Lokathor/safe_arch#77]
There's a few problems with adding AVX-512 support:
* I don't have a device to develop on that even supports AVX-512 (for testing).
* Within Rust, it's all Nightly-only.

So (for now) we're blocked on adding avx512, but it'd be nice to have "some day".

This wouldn't be a 1.0 blocker, if we got to a state where we were otherwise 1.0 ready.

**Comment by Wunkolo** [comment]
The [Intel software development emulator](https://software.intel.com/content/www/us/en/develop/articles/intel-software-development-emulator.html) will allow you to at least verify a proper AVX-512 implementation, though debugging may be difficult.

**Comment by Lokathor** [comment]
I'm gonna close this since it's been a year and avx-512 is not even fully on Nightly. we can always revisit later

**Comment by polarathene** [comment]
With the recent AMD Zen 4 release introducing AVX-512 support too, has much changed since the issue was closed?

- Is [this the issue to track for support on AVX-512](https://github.com/rust-lang/rust/issues/44839)?
- [`packed_simd_2`](https://github.com/rust-lang/packed_simd) is where the nightly support is being worked on? Which mentions that [`stdsimd` / `portable-simd`](https://github.com/rust-lang/portable-simd) will eventually replace it (_once stable I guess?_).

So I gather not much has changed, other than hardware availability improving, the support may continue to depend on nightly for a while?

---

For a device to test on, it might be possible to rent a VPS instance hourly as needed? I think some providers like Amazon also offer a free tier for a year to new sign-ups, which might make that viable? (_You can find [info on instance types here](https://aws.amazon.com/ec2/instance-types/), eg M6i has AVX-512_)

**Comment by Lokathor** [comment]
`packed_simd_2` exists only so that required bugfixes can be added. Specifically, the person that had publishing rights to `packed_simd` had left rust development for a while, so we had to fork it and release it under `packed_simd_2`. Other than that it's not developing as far as I know.

`portable-simd` is what's in active development, but that intentionally is not attempting to offer specific individual instrinsics.

If any place is going to allow specific individual intrinsics, that would be `stdarch` I think.
